<launch>
  <arg name="gui" default="true"/>


  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find risk_aware_planning)/gazebo/worlds/P22_IF.world"/>
    <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="$(arg gui)"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="false"/>
    <arg name="verbose" default="true"/>
  </include>

  <arg name="robot" default="youbot"/>
  <!-- <arg name="init_pos_x" default="5.6"/>
  <arg name="init_pos_y" default="3.8"/>
  <arg name="init_pos_z" default="0.1"/>
  <arg name="init_pos_yaw" default="3.14"/> -->
  <arg name="init_pos_x" default="2.9"/>
  <arg name="init_pos_y" default="6.1"/>
  <arg name="init_pos_z" default="0.1"/>
  <arg name="init_pos_yaw" default="0"/>
  <arg name="init_pos_j1" default="0.03"/>
  <arg name="init_pos_j2" default="0.5"/>
  <arg name="init_pos_j3" default="-2"/>
  <arg name="init_pos_g" default="0.01"/>

  <!-- launch world -->
  <param name="robot_description" command="$(find xacro)/xacro.py $(find youbot_description)/robots/$(arg robot).urdf.xacro" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_robot" respawn="false" output="screen"
    args="-param robot_description
          -urdf
          -x $(arg init_pos_x)
          -y $(arg init_pos_y)
          -z $(arg init_pos_z)
          -Y $(arg init_pos_yaw)
          -J arm_joint_1 $(arg init_pos_j1)
          -J arm_joint_2 $(arg init_pos_j2)
          -J arm_joint_3 $(arg init_pos_j3)
          -J gripper_finger_joint_l $(arg init_pos_g)
          -J gripper_finger_joint_r $(arg init_pos_g)
          -model youbot">
  </node>

  <!-- launch joint state controller -->
  <include file="$(find youbot_gazebo_control)/launch/joint_state_controller.launch" />

  <!-- launch base controller -->
  <include file="$(find youbot_gazebo_control)/launch/base_controller.launch" />

  <!-- launch arm controller -->
  <include file="$(find youbot_gazebo_control)/launch/arm_controller.launch" />

  <!-- launch gripper controller -->
  <include file="$(find youbot_gazebo_control)/launch/gripper_controller.launch" />

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="100.0" />
  </node>

  <node name="main" type="main_simulation.py" pkg="risk_aware_planning" output="screen">
  </node>
</launch>
